wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt

```diff
@@ -145,6 +149,8 @@ class WireCompiler internal constructor(
             .withAndroidAnnotations(emitAndroidAnnotations)
             .withCompact(emitCompact)
 
+        // TODO(oldergod) crashes if one tries to mix javaOut and services?
+        val types = ConcurrentLinkedQueue<Type>(queue.filter { it is Type } as List<Type>)
```

Comment:
http://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/filter-is-instance.html